bugsnag-android-core/src/main/java/com/bugsnag/android/Event.java

```diff
@@ -0,0 +1,325 @@
+package com.bugsnag.android;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An Event object represents a Throwable captured by Bugsnag and is available as a parameter on
+ * an {@link OnErrorCallback}, where individual properties can be mutated before an error report is
+ * sent to Bugsnag's API.
+ */
+@SuppressWarnings("ConstantConditions")
+public class Event implements JsonStream.Streamable, MetadataAware, UserAware {
+
+    final EventInternal impl;
```

Comment:
Shall we call this `_event` or something similar to indicate internal-ness?

Reply:
Agreed to leave this as-is since `_event` would go against our checkstyle rules.